File: 2079-watering-plants/2079-watering-plants.java
class Solution {
    public int wateringPlants(int[] p, int capacity) {
        int res = 0, can = capacity;
    for (int i = 0; i < p.length; ++i) {
        if (can < p[i]) {
            res += i * 2;
            can = capacity;
        }
        can -= p[i];
    }
    return res + p.length;    
    }
}

File: 2079-watering-plants/2079-watering-plants.cpp
class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) {
        int n=plants.size();
        if(plants[0]<=capacity&&n==1) return 1;
        int ans=0;
        int tem=capacity;
        for(int i=0;i<n;++i)
        {
            if(plants[i]<=capacity)
            {
                ans++;
                capacity-=plants[i];
            }
            else
            {
                
                capacity=tem;
                capacity-=plants[i];
                ans+=(2*i)+1;
            }
        }
        return ans;
    }
};